sumologic/resource_sumologic_s3_data_forwarding_rule.go

```diff
@@ -0,0 +1,112 @@
+package sumologic
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceSumologicS3DataForwardingRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSumologicS3DataForwardingRuleCreate,
+		Read:   resourceSumologicS3DataForwardingRuleRead,
+		Update: resourceSumologicS3DataForwardingRuleUpdate,
+		Delete: resourceSumologicS3DataForwardingRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"index_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"destination_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"file_format": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "{index}_{day}_{hour}_{minute}_{second}",
+			},
+			"format": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "csv",
+			},
+		},
+	}
+}
+
+func resourceSumologicS3DataForwardingRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	if d.Id() == "" {
+		dfr := resourceToS3DataForwardingRule(d)
+		createdDfd, err := c.CreateS3DataForwardingRule(dfr)
+
+		if err != nil {
+			return err
+		}
+
+		d.SetId(createdDfd.IndexID)
+	}
+
+	return resourceSumologicS3DataForwardingRuleRead(d, meta)
+}
+
+func resourceSumologicS3DataForwardingRuleRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+	id := d.Id()
+	dfr, err := c.GetS3DataForwardingRule(id)
+
+	if err != nil {
+		return err
+	}
+
+	if dfr == nil {
+		d.SetId("")
+		return fmt.Errorf("S3DataForwardingRule for id %s not found", id)
+	}
+
+	d.SetId(dfr.IndexID)
```

Comment:
The id of a data forwarding rule isn't equal to the indexId. Please set the id of the rule here